### PR TITLE
[improve][docs] Add schema compatibility between Go and Java client

### PR DESCRIPTION
**Pull request by liangyepianzhou**
original PR: https://github.com/apache/pulsar-client-go/pull/1056

- [x] `doc` 
- [ ] `doc-required` 
- [ ] `doc-not-needed` 
- [ ] `doc-complete`

### CODE REVIEW

**Comment by Anonymitaet** [comment]
@BewareMyPower @shibd 
Could you please help review this PR? Thank you!

File: docs/client-libraries-schema.md [hunk 1]
```diff
@@ -142,4 +142,96 @@ while True:
...
+Avro schema in Go and Java are compatible, but there are some differences in how the schemas are defined. Go typically uses schema definitions, a string JSON, to create schemas. However, Java often uses class types for schema creation. As a result, Java allows non-primitive fields to be nullable by default, while in Go's schema definition, the nullability of fields needs to be explicitly stated.
+GO:
+```go
+// Compatible with defining a schema in Java
+exampleSchemaDefCompatible := NewAvroSchema(`{"fields":
+    [
+        {"name":"id","type":"int"},{"default":null,"name":"name","type":["null","string"]}
+    ],
+    "name":"MyAvro","namespace":"schemaNotFoundTestCase","type":"record"}`, nil)
+// Not compatible with defining a schema in Java
```
Comment (shibd) [inline docs/client-libraries-schema.md:161]: Does this refer to the following Java example?

If the Go Client is used like this, the Java client should also support it, right? 

In java client, there are two ways:
1. Use SchemaDefinition to set schema JsonDef

``` 
          String jsonDef = "{"fields":[{"name":"id","type":"int"},{"default":null,"name":"name","type":["null","string"]}],"name":"MyAvro","namespace":"schemaNotFoundTestCase","type":"record"}"
           Schema schema = Schema.AVRO(SchemaDefinition.builder().withJsonDef("").build());
```

2. Or, to set not allow field is null.
```
      SchemaDefinition<Example> schemaDefinition =
                     SchemaDefinition.builder().withPojo(Example.class).withAlwaysAllowNull(false).build();
      Schema schema = Schema.AVRO(schemaDefinition);
      
    Producer<Example> producer = pulsarClient.newProducer(schema)
                .topic(topic).create();
```

we should show the solution.

File: docs/client-libraries-schema.md [hunk 2]
```diff
@@ -142,4 +142,96 @@ while True:
...
+
+#### JSON Schema
+The situation with JSON schema is similar to Avro Schema.
+```go
+// Compatible with defining a schema in Java
+exampleSchemaDefCompatible := "{\"type\":\"record\",\"name\":\"Example\",\"namespace\":\"test\"," +
+	"\"fields\":[{\"name\":\"ID\",\"type\":\"int\"},{\"name\":\"Name\",\"type\":[\"null\", \"string\"]}]}"
+
+consumerJSCompatible := NewJSONSchema(exampleSchemaDefCompatible, nil)
+// Not compatible with defining a schema in Java
```
Comment (shibd) [inline docs/client-libraries-schema.md:191]: Which define code of Java is this incompatible with?

Comment (liangyepianzhou) [reply docs/client-libraries-schema.md:191]: Like Avro, is a problem with null fields.

**Pull request closed without merging**